news/10002.feature.rst

```diff
@@ -0,0 +1,7 @@
+New resolver: Loosen URL comparison logic when checking for direct URL reference
+equivalency. The logic includes the following notable characteristics:
+
+* The authentication part of the URL is explicitly ignored.
+* Most of the fragment part, including ``egg=``, is explicitly ignored. Only
+  ``subdirectory=`` and hash values (e.g. ``sha256=``) are kept.
+* The query part of the URL is hashed to allow ordering differences.
```

Comment:
```suggestion
* The query part of the URL is normalised to allow ordering differences.
```

(The previous wording read to me as though the query string was run through a hashing function, which would not allow for ordering differences)